File: 2215-find-the-difference-of-two-arrays/main.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        
        set1 = set(nums1)
        set2 = set(nums2)
        
        diff_nums1 = list(set1 - set2)
        diff_nums2 = list(set2 - set1)
        
        return [diff_nums1, diff_nums2]

File: 2215-find-the-difference-of-two-arrays/main.rs
use std::collections::HashSet;

impl Solution {
    pub fn find_difference(nums1: Vec<i32>, nums2: Vec<i32>) -> Vec<Vec<i32>> {

        let set1: HashSet<i32> = nums1.into_iter().collect();
        let set2: HashSet<i32> = nums2.into_iter().collect();


        let diff1: Vec<i32> = set1.difference(&set2).cloned().collect();
        let diff2: Vec<i32> = set2.difference(&set1).cloned().collect();

        vec![diff1, diff2]
    }
}
